### PR TITLE
update website font, misc UI tweaks

### PR DESCRIPTION
**Pull request by Simek**
This PR updates the website default font stack to use `Optimistic Display` font, which lately has been used on the React website beta. The new font is used on the whole website, including Algolia DocSearch components. I have decided to use `light` variant as a base font for the content (which is not that thin), because with default weight `500` everything looks a bit overwhelming (it was harder to focus and differentiate the heading for example).

The changeset also includes few tweaks and fixes, mainly for the blog sidebar, blog layout and navbar. Rest of the changes is related to the change of default `line-height` of new font.

I think it's fine, for now, to declare this font inside SCSS files, but if there will be issues with font loading time for the visitors we can move font declaration to the `Head` tags (current setup mimics the one used on beta website).

# Preview

### macOS

Screenshot 2022-01-26 at 18 43 39

Screenshot 2022-01-26 at 18 58 44

### Windows
Screenshot 2022-01-26 184446
Screenshot 2022-01-26 190142

### CODE REVIEW

File: website/static/img/homepage/cross-platform.svg [hunk 1]
```diff
@@ -1 +1 @@
...037919-3.3693163 1.83629677v11.16721749c0 2.204163 1.5041128 3.0291031 3.3693163 1.8362968z" fill="#999"/></g><g transform="matrix(1 0 0 -1 323 365)"><path d="m0 8.5h1.49809915 27.50190085" stroke="#999" stroke-width="6"/><path d="m40.2835353 9.57993401c.9324575-.59631093.9356445-1.56108391 0-2.15943291l-9.914219-6.34018906c-1.860822-1.19000433-3.3693163-.37037919-3.3693163 1.83629677v11.16721749c0 2.204163 1.5041128 3.0291031 3.3693163 1.8362968z" fill="#999"/></g><g transform="matrix(1 0 0 -1 323 296)"><path d="m0 8.5h1.49809915 27.50190085" stroke="#999" stroke-width="6"/><path d="m40.2835353 9.57993401c.9324575-.59631093.9356445-1.56108391 0-2.15943291l-9.914219-6.34018906c-1.860822-1.19000433-3.3693163-.37037919-3.3693163 1.83629677v11.16721749c0 2.204163 1.5041128 3.0291031 3.3693163 1.8362968z" fill="#999"/></g><g transform="matrix(1 0 0 -1 323 432)"><path d="m0 8.5h1.49809915 27.50190085" stroke="#999" stroke-width="6"/><path d="m40.2835353 9.57993401c.9324575-.59631093.9356445-1.56108391 0-2.15943291l-9.914219-6.34018906c-1.860822-1.19000433-3.3693163-.37037919-3.3693163 1.83629677v11.16721749c0 2.204163 1.5041128 3.0291031 3.3693163 1.8362968z" fill="#999"/></g></g></svg>
```
Comment (lunaleaps) [inline website/static/img/homepage/cross-platform.svg:1]: What is this change for?

Comment (Simek) [reply website/static/img/homepage/cross-platform.svg:1]: Font change in SVG graph on the front page:
Screenshot 2022-01-26 at 19 32 39

Comment (lunaleaps) [reply website/static/img/homepage/cross-platform.svg:1]: Wow you really made sure all fonts were changed!

**Review by lunaleaps — APPROVED** [review]
Changes look good to me!

File: website/src/css/customTheme.scss [hunk 2]
```diff
@@ -64,12 +66,47 @@ html[data-theme="dark"] {
...
 
+::marker {
+  color: var(--ifm-font-color-secondary);
+}
+
+/* Font */
+
+@font-face {
+  font-family: "Optimistic Display";
+  src: url("https://facebookmicrosites.github.io/design/public/fonts/OptimisticDisplayLight-199be98cf48e5b4c688356b08a02362c.woff2")
```
Comment (lunaleaps) [inline website/src/css/customTheme.scss:77]: @rachelnabors  A non-blocking question -- is there a better host for these fonts? @Simek and I were wondering if there is a plan to host these on a CDN somewhere

**Pull request merged**